src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -4,12 +4,29 @@
 
 import org.junit.Test;
 
-public class TypewiseAlertTest 
-{
-    @Test
-    public void infersBreachAsPerLimits()
-    {
-      assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
-    }
+public class TypewiseAlertTest {
+
+	@Test
+	public void testCheckAndAlert() {
+		// assert alert for too high temperature - Email
+		BatteryCharacter batteryCharacter = new BatteryCharacter();
```

Comment:
Good set of tests 👍 
The verification also needs to be designed, in this case alert is sent to email or controller, but there is no feedback whether email or controller method has indeed been executed....

Reply:
Altered code little bit to check for response from controller or email. Kindly check if that's ok.

Reply:
Now it is verified that the right method is called  👍